lib/AST/RequirementMachine/RequirementLowering.cpp

```diff
@@ -108,9 +146,20 @@ static void desugarSuperclassRequirement(Type subjectType,
 
 static void desugarLayoutRequirement(Type subjectType,
                                      LayoutConstraint layout,
-                                     SmallVectorImpl<Requirement> &result) {
+                                     SourceLoc loc,
+                                     SmallVectorImpl<Requirement> &result,
+                                     SmallVectorImpl<RequirementError> &errors) {
   if (!subjectType->isTypeParameter()) {
-    // FIXME: Diagnose redundancy or conflict upstream
+    Requirement requirement(RequirementKind::Layout,
+                            subjectType, layout);
+    if (subjectType->isAnyClassReferenceType()) {
```

Comment:
This should be conditional on the requirement itself being a class layout requirement. You can't write any other kind of layout requirement in a normal generic signature, but you can in a `@_specialize`.

Reply:
Oh neat! Is there a way to check if a given type satisfies a given layout requirement, or should I just add the condition that the layout constraint is a class layout like you suggest?

Reply:
Unfortunately we don't have a way to statically check other layout requirements